### PR TITLE
add anti affinity to docker registry pods

### PR DESCRIPTION
**Pull request by gabemontero**
https://jira.coreos.com/browse/DEVEXP-249

@openshift/sig-developer-experience fyi

/hold  

still want to experiment with changing the replicas setting on the config object ... getting the PR up for a sanity check since I missed planning, plus get regression testing

### CODE REVIEW

**Comment by gabemontero** [comment]
test errors seemed mostly around odd authorization complaints ... let's see if there is any consistency in the failures

/retest

File: pkg/resource/podtemplatespec.go [hunk 1]
```diff
@@ -254,6 +254,21 @@ func makePodTemplateSpec(coreClient coreset.CoreV1Interface, driver storage.Driv
...
 			SecurityContext:    securityContext,
+			Affinity: &corev1.Affinity{
+				PodAntiAffinity: &corev1.PodAntiAffinity{
+					PreferredDuringSchedulingIgnoredDuringExecution: []corev1.WeightedPodAffinityTerm{
+						{
+							Weight: 100,
+							PodAffinityTerm: corev1.PodAffinityTerm{
+								//TODO godoc for this field says it cannot be empty, but the doc at
+								// https://kubernetes.io/docs/concepts/configuration/assign-pod-node/#an-example-of-a-pod-that-uses-pod-affinity
+								// talks about using an empty topologyKey with anti-affinity as signifying "all topologes"
```
Comment (coreydaley) [inline pkg/resource/podtemplatespec.go]: s/topologes/topologies/

**Comment by gabemontero** [comment]
Still getting the unauthorized to change users on the image reg e2e, but I also saw that in some openshfit/jenkins and openshift/origin PRs, so I'm still contending they are unrelated.

But if more experienced image registry dev's at @openshift/sig-developer-experience know better, please chime in here.

Did some local testing, and can confirm that when changing the operator to employ multiple replicas (where I put the operator in Removed state, then changed the replicas, then put back in Managed), the `image-registry-*` pods are showing up on different nodes.

removing the hold

/hold cancel

/assign @dmage

File: pkg/resource/podtemplatespec.go (hunk 1 above)
Comment (gabemontero) [reply pkg/resource/podtemplatespec.go]: update pushed - thanks

**Comment by gabemontero** [comment]
Yep saw the same auth error in @legionus 's https://github.com/openshift/cluster-image-registry-operator/pull/242

Something is amiss underneath the image reg e2e's

File: pkg/resource/podtemplatespec.go [hunk 2]
```diff
@@ -254,6 +254,22 @@ func makePodTemplateSpec(coreClient coreset.CoreV1Interface, driver storage.Driv
...
+				PodAntiAffinity: &corev1.PodAntiAffinity{
+					PreferredDuringSchedulingIgnoredDuringExecution: []corev1.WeightedPodAffinityTerm{
+						{
+							Weight: 100,
+							PodAffinityTerm: corev1.PodAffinityTerm{
+								//TODO godoc for this field says it cannot be empty, but the doc at
+								// https://kubernetes.io/docs/concepts/configuration/assign-pod-node/#an-example-of-a-pod-that-uses-pod-affinity
+								// talks about using an empty topologyKey with anti-affinity as signifying "all topologies",
+								// That said, the standard kubernetes.io/hostname has appeared sufficient in testing on AWS clusters with 3 worker nodes
+								TopologyKey: "kubernetes.io/hostname",
```
Comment (dmage) [inline pkg/resource/podtemplatespec.go]: You can use `params.Deployment.Namespace` to get the namespace.

Comment (gabemontero) [reply pkg/resource/podtemplatespec.go]: update pushed @dmage ... thanks

**Comment by gabemontero** [comment]
we'll see how e2e-aws-image-registry does on its current run, but others are seeing similar unauthorized hiccups as well .... see https://coreos.slack.com/archives/CEKNRGF25/p1553093413110100

**Comment by gabemontero** [comment]
Looks like https://bugzilla.redhat.com/show_bug.cgi?id=1688820 and https://bugzilla.redhat.com/show_bug.cgi?id=1690101 have been reported and have the same "unauthorized" flavor

going to hold this until they merge

/hold

**Comment by gabemontero** [comment]
well a test got kicked off ...will hold after that just in case we get lucky :-)

/hold cancel

**Comment by gabemontero** [comment]
aws ip quota pain this time wrt e2e-aws-image-registry

**Pull request merged**